### PR TITLE
[BYOC] support arbitrary input dims for add/mul/relu of dnnl c_src codegen

### PR DESCRIPTION
**Pull request by sunwayforever**
[BYOC] support arbitrary input dims for add/mul/relu of dnnl c_src codegen

### CODE REVIEW

File: src/runtime/contrib/dnnl/dnnl.cc [hunk 1]
```diff
@@ -44,6 +44,32 @@ typedef struct {
...
+      data_md = dnnl::memory::desc({shape, dtype, tag::abc});
+      break;
+    case 4:
+      data_md = dnnl::memory::desc({shape, dtype, tag::abcd});
+      break;
+    case 5:
+      data_md = dnnl::memory::desc({shape, dtype, tag::abcde});
+      break;
+    default:
+      assert(true);
```
Comment (comaniac) [inline src/runtime/contrib/dnnl/dnnl.cc:67]:
```suggestion
      LOG(FATAL) << "message";
```

**Review by comaniac — COMMENTED** [review]
Otherwise LGTM.

**Comment by masahi** [comment]
Can you add a test?

**Comment by comaniac** [comment]
> Can you add a test?

Hey @masahi, the DNNL C codegen currently is not enabled in the CI so it won't be tested anyways. The reason we didn't deprecate it is because it is a good codegen example against to the JSON codegen/runtime. Do you have any suggestion for this case?

**Review by masahi — APPROVED** [review]
I see, I think it's fine without a test then.

**Pull request merged**